### PR TITLE
'import' syntax compatible and TypeScript declaration file 

### PR DESCRIPTION
**Pull request by radetsky**
- Added compatibility with `import` syntax to `jsthemis` 
- TypeScript declaration file
- Increased the version of package.json
- Updated CHANGELOG.md 

## Checklist

- [x] Change is covered by automated tests
- [x] Changelog is updated (in case of notable or breaking changes)

### CODE REVIEW

File: src/wrappers/themis/jsthemis/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+export declare class KeyPair {
```
Comment (Lagovas) [inline src/wrappers/themis/jsthemis/index.d.ts:1]: did you write this file manually or somehow generated from the sources?

File: src/wrappers/themis/jsthemis/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    decrypt(message: Uint8Array): Uint8Array;
+}
+
+type GetPublicKeyCallback = (peerID: Uint8Array) => Uint8Array | null;
+
+export declare class SecureSession {
+    constructor(peerID: Uint8Array, privateKey: Uint8Array, getPublicKeyCallback: GetPublicKeyCallback);
+    isEstablished(): boolean;
+    connectRequest(): Uint8Array;
+    sendToPeer(message: Uint8Array): void;
```
Comment (Lagovas) [inline src/wrappers/themis/jsthemis/index.d.ts:48]: I didn't find these functions in secureSession.cpp to find result type. Where this function is declared?

File: src/wrappers/themis/jsthemis/index.d.ts (hunk 1 above)
Comment (radetsky) [reply src/wrappers/themis/jsthemis/index.d.ts:1]: Manually. I did not find the tool that can read `.node` file to make a declaration file.

File: src/wrappers/themis/jsthemis/index.d.ts (hunk 2 above)
Comment (radetsky) [reply src/wrappers/themis/jsthemis/index.d.ts]: Removed. Thank you. I did not write tests for SecureSession.

**Review by vixentael — APPROVED** [review]
@Lagovas looks good for you?

**Comment by vixentael** [comment]
Should we add tests for this file?

**Comment by radetsky** [comment]
> And I see, that you tried to add examples and update tests on CI but reverted changes. Did you met any problems with tests with new `import` syntax?

Nope, I met the issues with node >= 18. Decided to provide it later because of OpenSSL 3 issues.

File: src/wrappers/themis/jsthemis/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    encrypt(plaintext: Uint8Array, context?: Uint8Array): Uint8Array;
+    decrypt(encrypted: Uint8Array, context?: Uint8Array): Uint8Array;
+}
+
+export declare interface SecureCellTokenProtectResult {
+    data: Uint8Array;
+    token: Uint8Array;
+}
+
+export declare class SecureCellTokenProtect {
```
[the remark this answers is not in the document]
Comment (radetsky) [reply src/wrappers/themis/jsthemis/index.d.ts:22]: I've checked. We do not announce that we can do it, but construction with `new SecureCellTokenProtect(masterkey)` will also work the same as `SecureCellTokenProtect.withKey(masterKey)`. It wraps one function into a constructor. 
I'm adding the constructor declaration.

**Comment by Lagovas** [comment]
> Nope, I met the issues with node >= 18. Decided to provide it later because of OpenSSL 3 issues.

can we test it for openssl1.0/node <= 18?

File: docs/examples/ts/secure_cell.js
Comment (Lagovas) [inline docs/examples/ts/secure_cell.js:1]: do we need this file here? looks like it's generated file from the .ts

File: .github/workflows/test-nodejs.yaml [hunk 4]
```diff
@@ -132,3 +133,62 @@ jobs:
           node secure_comparator_client.js
           kill -SIGTERM "$!"
           echo "ok"
+      - name: Test 'import' syntax
+        if: always()
+        run: |
+          cd $GITHUB_WORKSPACE/docs/examples/js/
+          echo "Test import syntax..."
+          node import_module.mjs
```
Comment (Lagovas) [inline .github/workflows/test-nodejs.yaml:141]: just to be sure, did you tried some failed script to check some failure case to be sure that it's not always `echo ok`?

I found documentation for [run](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell) where described that by default `shell` used with `-e` flag. But real world can be unsynchronized from documentation and we met situations when tests ran but didn't exit with errors on failures.

Comment (radetsky) [reply .github/workflows/test-nodejs.yaml:141]: Screenshot 2023-11-28 at 12 55 54
Is it enough? Should I do a unit test?

**Review by Lagovas — APPROVED** [review]
lgtm. merge after removing redundant .js file as discussed

**Pull request merged**